test/CDPSession.spec.ts

```diff
@@ -88,6 +88,19 @@ describeChromeOnly('Target.createCDPSession', function () {
     }
     expect(error.message).toContain('Session closed.');
   });
+  it('should expose other sessions', async () => {
+    const { page, server } = getTestState();
+
+    const client = await page.target().createCDPSession();
+    const otherSessions = [];
+    client
+      .connection()
+      .on('sessionattached', (session) => otherSessions.push(session));
+
+    // This won't work yet.
```

Comment:
I had trouble locating the tests for the connection class. This test really needs OOPIFs in order to work which are quite complicated to force locally. Any ideas or thoughts on a test that launches Chrome with a complex set of flags and fixtures?

Reply:
Could we just launch a secondary server on a different port?

Reply:
In our testing that's not quite enough to trigger an OOPIF reliably.

@paulirish explored a self-contained repro in https://github.com/GoogleChrome/lighthouse/pull/11921 that just requires some Chrome flags that mess with host resolver rules. 

> basic approach
run chrome with --host-rules="MAP oopifdomain 127.0.0.1" --isolate-origins="http://oopifdomain:10503" and then open an iframe to "http://oopifdomain:10503"....

I was a little unclear how Chrome is launched as part of these tests though. If anyone could be so kind as to point me in the right direction I can give this a whirl :)

Reply:
The test would work only in headful mode. It should be moved to `test/headful.spec.ts`.